### PR TITLE
WebDriver: Add Get All Cookies Endpoint

### PR DESCRIPTION
**Pull request by TobyAsE**
Another tick for issue https://github.com/SerenityOS/serenity/issues/15551 @AtkinsSJ

### CODE REVIEW

File: Userland/Services/WebDriver/Session.cpp [hunk 1]
```diff
@@ -158,4 +160,46 @@ ErrorOr<JsonValue, HttpError> Session::get_title()
     return JsonValue(m_browser_connection->get_title());
 }
 
+// https://w3c.github.io/webdriver/#dfn-serialized-cookie
+static JsonObject serializeCookie(Web::Cookie::Cookie cookie)
```
Comment (linusg) [inline Userland/Services/WebDriver/Session.cpp:164]: snek_case

Comment (linusg) [reply Userland/Services/WebDriver/Session.cpp:164]: Also, can we pass the cookie by const reference?

File: Userland/Services/WebDriver/Session.cpp [hunk 2]
```diff
@@ -158,4 +160,46 @@ ErrorOr<JsonValue, HttpError> Session::get_title()
...
+{
+    JsonObject serializedCookie = {};
```
Comment (linusg) [inline Userland/Services/WebDriver/Session.cpp:166]: snek_case

File: Userland/Services/WebDriver/Session.cpp [hunk 3]
```diff
@@ -158,4 +160,46 @@ ErrorOr<JsonValue, HttpError> Session::get_title()
...
+    if (!current_window.has_value())
+        return HttpError { 404, "no such window", "Window not found" };
+
+    // FIXME: 2. Handle any user prompts, and return its value if it is an error.
+
+    // 3. Let cookies be a new JSON List.
+    JsonArray cookies = {};
+
+    // 4. For each cookie in all associated cookies of the current browsing context’s active document:
+    for (Web::Cookie::Cookie cookie : m_browser_connection->get_all_cookies()) {
```
Comment (linusg) [inline Userland/Services/WebDriver/Session.cpp:193]: `auto const&`

File: Userland/Services/WebDriver/Session.cpp [hunk 4]
```diff
@@ -158,4 +160,46 @@ ErrorOr<JsonValue, HttpError> Session::get_title()
...
+        // 1. Let serialized cookie be the result of serializing cookie.
+        auto serializedCookie = serializeCookie(cookie);
```
Comment (linusg) [inline Userland/Services/WebDriver/Session.cpp:195]: snek_case

**Review by linusg — CHANGES_REQUESTED** [review]

**Review by linusg — APPROVED** [review]

**Comment by AtkinsSJ** [comment]
Looks like merging your other PR gave this a conflict. (Been there, done that. :laughing:)

**Comment by TobyAsE** [comment]
@AtkinsSJ  All resolved now :^)

**Comment by linusg** [comment]
Conflicts :sweat_smile:

**Comment by TobyAsE** [comment]
And resolved again.
Man, the diff algorithm really doesnt do a great job untangling the commits...

File: Userland/Services/WebDriver/Session.cpp [hunk 5]
```diff
@@ -229,4 +231,46 @@ ErrorOr<JsonValue, HttpError> Session::forward()
     return JsonValue();
 }
 
+    // https://w3c.github.io/webdriver/#dfn-serialized-cookie
```
Comment (linusg) [inline Userland/Services/WebDriver/Session.cpp:234]: Botched comment

Comment (TobyAsE) [reply Userland/Services/WebDriver/Session.cpp]: Eagle eyes!
Fixed

**Pull request merged**